sector-builder/src/helpers/snapshots.rs

```diff
@@ -20,80 +37,106 @@ pub fn load_snapshot<T: KeyValueStore>(
     Ok(None)
 }
 
+impl From<&SnapshotKey> for Vec<u8> {
+    fn from(n: &SnapshotKey) -> Self {
+        // convert the sector size to a byte vector
+        let mut snapshot_key = vec![];
```

Comment:
this could be `Vec::with_capacity(n.prover_id.len() + 8)`

Reply:
good call